src/App.js

```diff
@@ -59,9 +60,11 @@ function App() {
 function AppWithContext() {
   return (
     <ThemeProvider theme={theme}>
-      <ReduxProvider store={store}>
-        <App />
-      </ReduxProvider>
+      <StateInspector name="App">
+        <StoreProvider reducer={rootReducer}>
```

Comment:
The fact we're combining multiple reducers into a root reducer and piping that through a single context provider, leaves us in no different place performance wise than simply adding more reducers to the existing `MakerContextProvider` as we are already doing across the app, ultimately any solution which puts react context thru constant updates, especially in a centralized tree approach, will not accomplish the desired performance without digging into the bitwise optimizations at the `createContext` level.

That's important to consider so we're aware of potentially going thru all this to end up at the same initial problem and with 2 "same same but different" implementations with the same pitfalls while assuming different.

as much as I'm ok merging this is so to move along with the project progress and get over this roadblock, we're not solving the real limitations which we admit our current state management approach has with this implementation per se, we're rebuilding redux in react context.

I very much like what was proposed a few days ago by @levity about breaking down how/where our components consume state so to minimize re-renders (but really also to minimize the impact of our experimental state management solution across the app experience), and personally think that, in react 2019 lango, means breaking down into multiple, single purpose `ContextProviders`, which can within, optionally, have some of the attributes proposed here, like immutable data.

As for a proper bailout out of react land for us to do all the side-effect state management magic web3 dapps require to produce a good UX, that's a conversation we still need to have, and a decision we can't keep running away from.
while having that conversation, using our existing foundation as starting point, we can go thru some options and opt with the most pragmatic one, and we can of course talk about why redux is or isn't the best candidate for that.

Myself, I'd like to avoid "all or nothing" approaches which makes working with hooks harder than it already is.

Reply:
> multiple, single purpose ContextProviders

do you mean multiple single purpose `useReducers`? to me that's one of the most interesting ways in which we can productively depart from the pattern that Redux locked us into--we can have reducers local to just a part of the component tree that manage state change when it gets complex, without needing to put all of them into a global context